content/learning-paths/cross-platform/vectorization-friendly-data-layout/a-more-complex-problem.md

```diff
@@ -56,41 +63,51 @@ void simulate_objects(object_t *objects, float duration, float step) {
 }
 ```
 
-This is a scalar approach, but the code still uses the `vec4` struct and even the boundaries checking is done on the fourth element, isn't this a waste of cycles? Not really, because the compiler can find an opportunity to autovectorize this, but you should be able to verify this for yourself.
+This is a scalar approach, but the code still uses the `vec4` struct and the boundary checking is even done on the fourth element. Is this a waste of cycles? No, because the compiler can find an opportunity to autovectorize this, but you should be able to verify this for yourself.
 
-Compile this file first with `-O2`:
+First, compile the code with `-O2`:
 
-```bash
+```console
 gcc -O2 -Wall simulation2.c -o simulation2
 ```
 
-Running this should give the following:
+Run the new executable:
+
+```console
+ ./simulation2
+```
+
+The resulting output is:
 
-```bash
-$ ./simulation2
+```output
 Total border collisions: x: 250123, y: 249711, z: 249844
 elapsed time: 36.700929
 ```
 
-And now with `-O3`:
+Next, compile with `-O3`:
 
-```bash
+```console
 gcc -O3 -Wall simulation2.c -o simulation2
 ```
 
-And similarly you should get the following result:
+Run agin:
+
+```console
+./simulation2
+```
+
+Similar to last time, the output is:
 
-```bash
-$ ./simulation2
+```output
 Total border collisions: x: 250123, y: 249711, z: 249844
 elapsed time: 28.926221
 ```
 
-So a reduction in execution time of ~21%. This is decent, but you might probably be expecting more. Checking the assembly output will tell you that it's not as good as you expected. 
+Using `-O3` reduces the execution time by about 21%. This is good, but you might be expecting more. Checking the assembly output will tell you that it's not as good as you expected. 
 
-Check the [output of `objdump -S` for `-O2`](../simulate_objects_O2.s). It doesn't show any SIMD instructions, which is expected with `-O2`.
+Check the [assembly output of objdump -O2](../simulate_objects_O2.s). It doesn't show any SIMD instructions, which is expected with `-O2`.
```

Comment:
this should probably be `assembly output for -O2` as below.

Reply:
changed to be consistant